arm-datalake-analytics/job/2016-03-20-preview/swagger/job.json

```diff
@@ -0,0 +1,1017 @@
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the severity level of the failure.",
+          "enum": [
+            "Warning",
+            "Error",
+            "Info"
+          ],
+          "x-ms-enum": {
+            "name": "SeverityTypes",
+            "modelAsString": false
+          }
+        },
+        "source": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the ultimate source of the failure (usually either SYSTEM or USER)."
+        },
+        "startOffset": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int32",
+          "description": "Gets the start offset in the job where the error was found"
+        }
+      },
+      "description": "The Data Lake Analytics job error details."
+    },
+    "JobInnerError": {
+      "properties": {
+        "diagnosticCode": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int32",
+          "description": "Gets the diagnostic error code."
+        },
+        "severity": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the severity level of the failure.",
+          "enum": [
+            "Warning",
+            "Error",
+            "Info"
+          ],
+          "x-ms-enum": {
+            "name": "SeverityTypes",
+            "modelAsString": false
+          }
+        },
+        "details": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the details of the error message."
+        },
+        "component": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the component that failed."
+        },
+        "errorId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the specific identifier for the type of error encountered in the job."
+        },
+        "helpLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the link to MSDN or Azure help for this type of error, if any."
+        },
+        "internalDiagnostics": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the internal diagnostic stack trace if the user requesting the job error details has sufficient permissions it will be retrieved, otherwise it will be empty."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the user friendly error message for the failure."
+        },
+        "resolution": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the recommended resolution for the failure, if any."
+        },
+        "source": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets the ultimate source of the failure (usually either SYSTEM or USER)."
```

Comment:
Should this be an enum?
If you think this will have more values, then you can have modelAsString: true for the x-ms-enum. This will atleast provide intellisense for strongly typed languages.